### PR TITLE
Add Eclipse Collections as a dependency.

### PR DESCRIPTION
**Pull request by itohiro73**
There are couple of open issues I want to consult with @gs-rezaem . Please kindly advise.

- In #30 , Moh suggested to try getting rid of GSC impl dependency, but there are too many dependencies for `FastList`/`UnifiedSet`/`UnifiedMap` as internal storage, so I left them as is for now.
- In #30, Moh suggested below for `ByteArraySet`, but this caused conflict in get(Object) and get(K), also signature conflict in collect(). So I didn't touch this class for now. Please advise how you want to handle this class.
> For ByteArraySet, extend the EC UnifiedSetWithHS and implement GSC MutableMap.
- `MithraFastList` and `AdhocFastList` extend `FastList`. Please advise how you want to handle these classes. 
- `AbstractDatedCache.MatchAllAsOfDatesProcedureForMany#getResult()` returns `FastList`. Please advise how you want to handle this.
- `MasterSyncResult#getBuffers()` returns `FastList`. Please advise how you want to handle this.
- `MithraCompositeList#getLists()` returns `FastList`. Please advise how you want to handle this.
- `AsOfEqualityChecker#getMsiPool()` returns `UnifiedMap`. Please advise how you want to handle this.
- `ConcurrentIntObjectHashMap#parallelForEachValue()` takes `List<IntObjectProcedure<V>>` in its signature. Introducing EC equivalent method would cause same erasure issue. Please advise how you want to handle this.
- `ListFactory` has couple of create() methods that return `FixedSizeList`. We can either have GSC version or EC version for this class. Making it EC version may cause wide impact as it's used widely in reladomo. Please advise how you want to handle this class.

### CODE REVIEW

**Comment by gs-rezaem** [comment]
I think replacing the internal usages is pretty important because it'll flush out some of the interactions with the pseudo-public methods (see below) that you're having trouble with.

For ByteArraySet, use the same solution as primitives: create a new one and deprecate the old. Use the com.gs.reladomo.util package for the new one.

The other methods you've listed is what I would call "pseudo-public". Convert all of them to EC.

**Comment by itohiro73** [comment]
Thanks @gs-rezaem for your comment! It is very helpful.

I've added a commit to replace GSC impl with EC impl. Now only 3 impl usages are left after the change. Because of these, we still cannot get rid of GSC lib dependency itself.

- `ListAbstract.jsp` for `asGscList()`
- `AggregateList#getAttributeAsGscIntSet(String)` that constructs `IntHashSet` in it
- Old `ByteArraySet` that implements GSC `UnifiedSetWithHashingStrategy`.

FYI, along with converting GSC impl to EC, I've optimized imports as well, so it may look more changes than necessary in imports.

Please let me know for further comments or suggestions.

**Comment by itohiro73** [comment]
Fixed issues in hiroshi_ito.dco commit.

- Fix file name typo: `hiroshi_ito.cdo` => `hiroshi_ito.dco`
- Fix missing replacement: `<your name>.dco` => `hiroshi_ito.dco`

**Comment by gs-rezaem** [comment]
asGscList is in the generated code and it's optional, so it's not a real dependency.
The other two are so rare in production, that I wonder if people would mind fixing a couple of places in their code when upgrading.
Can you put together a guide to changing a project over from GSC to EC? I can think of two steps:

- replace imports
- replace asGscList with asEcList

**Comment by itohiro73** [comment]
Sure, let me put together a guide.

As for two remaining GSC dependencies, do you want me to get rid of them or leave them as is for now?

**Comment by gs-rezaem** [comment]
Let's remove the two.

**Comment by itohiro73** [comment]
Now all GSC impl dependencies have been removed 👍.

I added a guide (`GSC_TO_EC_MIGRATION_GUIDE.md`) to migrate from GSC to EC, too. I'd assume there is going to be a link to this guide from `CHANGELOG.md` as part of 1.7.0 release.

Please let me know for any feedback.

File: GSC_TO_EC_MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+```
+```
+    <reladomo-gen xml="PATH_TO_THE_XML"
+        generatedDir="PATH_TO_THE_GEN_DIR"
+        nonGeneratedDir="PATH_TO_THE_NON_GEN_DIR"
+        generateEcListMethod="true" />
+```
+
+2. Replace `asGscList()` with `asEcList()` in your code. 
+3. Replace `com.gs.collections.*` imports with `eclipse.collections.*` in your code
```
Comment (donraab) [inline GSC_TO_EC_MIGRATION_GUIDE.md]: I think this should be 
Replace `com.gs.collections.*` imports with `org.eclipse.collections.*` in your code

Comment (itohiro73) [reply GSC_TO_EC_MIGRATION_GUIDE.md]: @donraab Good catch! I'll fix it.

Comment (itohiro73) [reply GSC_TO_EC_MIGRATION_GUIDE.md]: Fixed!

**Comment by gs-rezaem** [comment]
Merged with some minor changes. Thanks for this huge piece of work :smile:

**Pull request merged**